src/os/ui/layer/compare/layercompareui.js

```diff
@@ -462,6 +466,12 @@ const windowId = 'compare-layers';
  * @param {!LayerCompareOptions} options The layer compare options.
  */
 const launchLayerCompare = (options) => {
+  const featureCount = countFeatures(options.left) + countFeatures(options.right);
+  if (featureCount > getMaxFeatures('2d')) {
+    AlertManager.getInstance().sendAlert('Switching to 2D mode with the current data volume may degrade performance ' +
```

Comment:
Wording this as "switching to 2D mode" doesn't make sense for the layer compare since you aren't switching modes, its simply the mode we support for it. This should be something along the lines of "Using Layer Compare with the current data volume..."